docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1232 @@
++-----------------+------------------------------------------------------------------------------------+
+| ``ATTACHMENT``  | :class:`~discord.Attachment`                                                       |
++-----------------+------------------------------------------------------------------------------------+
+
+:ddocs:`Application command option types <interactions/application-commands#application-command-object-application-command-option-type>` as documented by Discord.
+
+User parameter
+^^^^^^^^^^^^^^^
+
+Annotating to either :class:`discord.User` or :class:`discord.Member` both point to a ``USER`` Discord-type.
+
+The actual type given by Discord is dependent on whether the command was invoked in DM-messages or in a guild.
+
+For example, if a parameter annotates to :class:`~discord.Member`, and the command is invoked in a guild,
+discord.py will raise an error since the actual type given by Discord,
+:class:`~discord.User`, is incompatible with :class:`~discord.Member`.
+
+discord.py doesn't raise an error for the other way around, ie. a parameter annotated to :class:`~discord.User` invoked in a guild -
+this is because :class:`~discord.Member` is compatible with :class:`~discord.User`.
+
+To accept member and user, regardless of where the command was invoked, place both types in a :obj:`~typing.Union`:
+
+.. code-block:: python
+
+    from typing import Union
+
+    @client.tree.command()
+    async def userinfo(
+        interaction: discord.Interaction,
+        user: Union[discord.User, discord.Member]
+    ):
+        info = user.name
+
+        # add some extra info if this command was invoked in a guild
+        if isinstance(user, discord.Member):
+            joined = user.joined_at
+            if joined:
+                relative = discord.utils.format_dt(joined, "R")
+                info = f"{info} (joined this server {relative})"
+
+        await interaction.response.send_message(info)
+
+Command groups
+---------------
+
+To make a more organised and complex tree of commands, Discord implements command groups and subcommands.
+A group can contain up to 25 subcommands or subgroups, with up to 1 level of nesting supported.
+
+Meaning, a structure like this is possible:
+
+.. code-block::
+
+    todo
+    ├── lists
+    │   ├── /todo lists create
+    │   └── /todo lists switch
+    ├── /todo add
+    └── /todo delete
+
+Command groups **are not invocable** on their own.
+
+Therefore, instead of creating a command the standard way by decorating an async function,
+groups are created by using :class:`.app_commands.Group`.
+
+This class is customisable by subclassing and passing in any relevant fields at inheritance:
+
+.. code-block:: python
+
+    class Todo(app_commands.Group, name="todo", description="manages a todolist"):
+        ...
+
+    client.tree.add_command(Todo()) # required!
+
+.. note::
+
+    Groups need to be added to the command tree manually with :meth:`.CommandTree.add_command`,
```

Comment:
What about [GroupCogs](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html?highlight=groupcog#discord.ext.commands.GroupCog)

Also within a cog they are also automatically added 
https://gist.github.com/AbstractUmbra/a9c188797ae194e592efe05fa129c57f#file-07-extension_with_optional_group-py

Or are Cogs out of scope for this guide?

Reply:
Okay, then I agree with the original phrasing and this feedback can be set to resolved 👍